userspace/libscap/engine/gvisor/scap_gvisor.cpp

```diff
@@ -239,6 +309,25 @@ int32_t engine::stop_capture()
 	shutdown(m_listenfd, 2);
 	::close(m_epollfd);
 	free_sandbox_buffers();
+
+	runsc_result sandboxes_res = runsc_list();
```

Comment:
Can't we instead cache the sandboxes list in `start_capture`?
i mean, we could be destroying sessions for unmanaged sandboxes here? Perhaps i miss some key pieces :)

Reply:
I believe caching doesn't work here, other sandboxes may be created after starting the capture and they will all be configured to connect to our socket, so each of them will have a session that must be deleted, if I understand correctly the question

Reply:
Makes sense; as i told you, i am missing some core key pieces about gvisor inner workings :)